sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/NoBackupContextWrapper.kt

```diff
@@ -0,0 +1,41 @@
+package cash.z.ecc.android.sdk.internal
+
+import android.content.Context
+import android.content.ContextWrapper
+import java.io.File
+
+/**
+ * Context class wrapped used for building our database classes. The advantage of this implementation
+ * is that we can control actions run on this context class.
+ *
+ * @param context
+ * @param parentDir The directory in which is the database file placed.
+ * @return Wrapped context class.
+ */
+internal class NoBackupContextWrapper(
+    context: Context,
+    private val parentDir: File?
```

Comment:
Should we make this non-null to simplify the implementation?

Reply:
@ccjernigan Yes, we can. But then we have to move the null check responsibility to the callers of this component, i.e. `DefaultSynchronizerFactory.defaultBlockStore` and `DatabaseCoordinatorKt.databaseBuilderNoBackupContext`, so my thought was to handle it on one single place -  `NoBackupContextWrapper`. There can be a null inputed, because we are working with `someFile.parentFile` field. What do you think about it?

Reply:
Here I ended up with checking the nullability on callers side and used the existing `DatabasePathException` in case of null parent, as we do on other places too.